plugins/scaffolder-backend/src/scaffolder/actions/builtin/filesystem/rename.test.ts

```diff
@@ -62,57 +64,47 @@ describe('fs:rename', () => {
     });
   });
 
-  it('should throw an error when files is not an array', async () => {
-    await expect(
-      action.handler({
-        ...mockContext,
-        input: { files: undefined } as any,
-      }),
-    ).rejects.toThrow(/files must be an Array/);
-
-    await expect(
-      action.handler({
-        ...mockContext,
-        input: { files: {} } as any,
-      }),
-    ).rejects.toThrow(/files must be an Array/);
-
-    await expect(
-      action.handler({
-        ...mockContext,
-        input: { files: '' } as any,
-      }),
-    ).rejects.toThrow(/files must be an Array/);
-
-    await expect(
-      action.handler({
-        ...mockContext,
-        input: { files: null } as any,
-      }),
-    ).rejects.toThrow(/files must be an Array/);
+  it('should throw ZodError when files is not an array', async () => {
+    const testCases = [
+      { input: undefined },
+      { input: {} },
+      { input: '' },
+      { input: null },
+    ];
```

Comment:
Using `Promise.all` within a Jest test isn't a common pattern. There's much nicety in going with Jest's `test.each` template literal I think. As in the 2nd form introduced in this [article](https://vanslaars.io/articles/jest-testing-with-test-each/).

There's some usage of the first form in the codebase, but I find the 2nd form to be very descriptive of what the test want, since we want to keep them DAMP 👍